pkg/common/config/aws_config.go

```diff
@@ -26,6 +46,59 @@ var (
 )
 
 func InitAWSViper() {
+	_ = viper.BindEnv(AWSSharedCredentials, "AWS_SHARED_CREDENTIALS")
+	RegisterSecret(AWSSharedCredentials, "aws-shared-credentials")
+
+	viper.SetDefault(AWSCredentialsFile, "/tmp/.aws/osde2e/credentials")
+	_ = viper.BindEnv(AWSCredentialsFile, "AWS_CREDENTIAL_FILE")
+
+	if viper.GetString(AWSSharedCredentials) != "" {
+		// If shared credntials file is provided in env vars, it should contain two profiles named "profile-a" and profile-b"
+		// Osde2e will use one of them based on current week.
+		// While one profile is in use, the other is cleaned up using AWS nuke
+		err := os.MkdirAll(filepath.Dir(viper.GetString(AWSCredentialsFile)), os.FileMode(0o755))
+		if err != nil {
+			log.Printf("Could not write given shared credentials file: %v", err)
+			return
+		}
+
+		data, err := base64.StdEncoding.DecodeString(viper.GetString(AWSSharedCredentials))
+		if err != nil {
+			log.Printf("Could not decode given shared credentials file. Ensure it is a valid base64 with no line breaks or spaces: %v", err)
+			return
+		}
+
+		// Write the string to the file
+		err = os.WriteFile(viper.GetString(AWSCredentialsFile), data, os.ModePerm)
+		if err != nil {
+			log.Printf("Could not write given shared credentials file: %v", err)
+			return
+		}
+
+		// use profile based on week
+		week := getWeekSince2024()
+		const profileA = "profile-a"
+		const profileB = "profile-b"
+		currentProfile := ""
+		if week%2 == 0 {
+			currentProfile = profileA
+		} else {
+			currentProfile = profileB
+		}
+		// remove secrets set in environment so that profile can take effect
+		// by default, AWS gives higher precedence to secret env vars than profile.
+		os.Setenv("AWS_ACCESS_KEY_ID", "")
+		os.Setenv("AWS_SECRET_ACCESS_KEY", "")
+		os.Setenv("AWS_ACCOUNT_ID", "")
+		os.Setenv("AWS_SHARED_CREDENTIALS_FILE", viper.GetString(AWSCredentialsFile))
+		os.Setenv("AWS_PROFILE", currentProfile)
+		_ = viper.BindEnv(AWSProfile, "AWS_PROFILE")
+		RegisterSecret(AWSProfile, "aws-profile")
+	} else {
+		_ = viper.BindEnv(AWSProfile, "AWS_PROFILE")
+		RegisterSecret(AWSProfile, "aws-profile")
+	}
```

Comment:
```suggestion
        }
	_ = viper.BindEnv(AWSProfile, "AWS_PROFILE")
	RegisterSecret(AWSProfile, "aws-profile")
```

it seems these are set to the same thing in both cases